azext_edge/tests/helpers.py

```diff
@@ -45,25 +45,33 @@ def filter_resources(
 def find_extra_or_missing_names(
     resource_type: str,
     result_names: List[str],
-    expected_names: List[str],
+    pre_expected_names: List[str],
+    post_expected_names: List[str],
     ignore_extras: bool = False,
-    # TODO: remove once dynamic pods check logic is implemented
     ignore_missing: bool = False,
 ):
+    """
+    Checks the result to find missing or extra names.
+    First checks the result name against the names in the kubectl results from before the result was fetched.
+    If anything is missing/extra, checks against the kubectl results from after the result was fetched.
```

Comment:
because of `_compare_support_bundle_names`, it might make sense to split this into two functions:
1. one function to generate two lists (missing vs extra names)
2. one to take and generate the error message from function 1

This should allow for `_compare_support_bundle_names` to just rely on function 1 while checks can still rely on function 1 and function 2